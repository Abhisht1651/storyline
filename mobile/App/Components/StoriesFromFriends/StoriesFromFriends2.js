// @flow

import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import Images from '../../Themes/Images'
import styles from '../Styles/StoriesFromFriendsStyles/StoriesFromFriends2Style'

type Props = {
  image1: Image,
  image2: Image,
  name: string,
  onPress: () => {}
}

const StoriesFromFriends2 = (props: Props) => {
  const { image1, image2, name, onPress } = props
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      <LinearGradient
        colors={['#46C3F4', '#B27CF3']}
        style={styles.linearGradient}
        start={{ x: 1, y: 0 }}
        end={{ x: 1, y: 1 }}>
        <TouchableOpacity style={styles.profilePicture} activeOpacity={1}>
          <Image source={image1} style={styles.image} />
        </TouchableOpacity>
      </LinearGradient>

      <View style={styles.TriangleShapeCSS} />
      <TouchableOpacity activeOpacity={1} style={styles.positon}>
        <LinearGradient
          colors={['#46C3F4', '#B27CF3']}
          style={styles.storyIcon}
          start={{ x: 1, y: 0 }}
          end={{ x: 1, y: 1 }}>
          <View style={styles.smallImageContainer}>
            <Image source={image2} style={styles.storyImage} />
          </View>
        </LinearGradient>
      </TouchableOpacity>
      <Text style={styles.text}>{name}</Text>
    </TouchableOpacity>
  )
}

StoriesFromFriends2.defaultProps = {
  image1: Images.flagPicture,
  image2: Images.profilePicture3,
  name: 'Name'
}

export default StoriesFromFriends2
