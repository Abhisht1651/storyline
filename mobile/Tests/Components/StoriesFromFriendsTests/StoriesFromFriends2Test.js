import { shallow } from 'enzyme'
import React from 'react'
import StoriesFromFriends2 from '../../../App/Components/StoriesFromFriends/StoriesFromFriends2'

describe('StoriesFromFriends2', () => {
  it('should match to snapshot', () => {
    const component = shallow(<StoriesFromFriends2 />)
    expect(component).toMatchSnapshot()
  })
  it('should render Image', () => {
    const component = shallow(<StoriesFromFriends2 />)
    expect(component.find('Image')).toHaveLength(2)
  })
})
