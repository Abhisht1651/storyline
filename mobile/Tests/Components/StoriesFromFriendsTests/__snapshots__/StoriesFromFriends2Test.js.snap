// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoriesFromFriends2 should match to snapshot 1`] = `
<ForwardRef
  style={
    Object {
      "alignItems": "center",
      "height": 208.64285714285714,
      "justifyContent": "space-between",
      "width": 118.28571428571428,
    }
  }
>
  <LinearGradient
    colors={
      Array [
        "#46C3F4",
        "#B27CF3",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 1,
      }
    }
    start={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    style={
      Object {
        "borderRadius": 24.642857142857142,
        "height": 152.78571428571428,
        "width": 119.92857142857143,
      }
    }
  >
    <ForwardRef
      activeOpacity={1}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 21.357142857142858,
          "flex": 1,
          "justifyContent": "center",
          "margin": 2,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "borderColor": "white",
            "borderRadius": 21.357142857142858,
            "borderWidth": 2,
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </ForwardRef>
  </LinearGradient>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderBottomColor": "#B27CF3",
        "borderBottomWidth": 11.5,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 6.571428571428571,
        "borderRadius": 2,
        "borderRightColor": "transparent",
        "borderRightWidth": 6.571428571428571,
        "borderStyle": "solid",
        "bottom": 36.964285714285715,
        "height": 0,
        "left": 72.28571428571428,
        "position": "absolute",
        "transform": Array [
          Object {
            "rotate": "19deg",
          },
        ],
        "width": 0,
      }
    }
  />
  <ForwardRef
    activeOpacity={1}
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <LinearGradient
      colors={
        Array [
          "#46C3F4",
          "#B27CF3",
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      start={
        Object {
          "x": 1,
          "y": 0,
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "borderRadius": 50,
          "height": 59.14285714285714,
          "top": 115,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 100,
            "flex": 1,
            "margin": 2,
          }
        }
      >
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "aspectRatio": 1,
              "borderRadius": 50,
              "height": "100%",
            }
          }
        />
      </View>
    </LinearGradient>
  </ForwardRef>
  <Text
    style={
      Object {
        "color": "#8F8FAF",
        "fontFamily": "BananaGrotesk-Regular",
        "fontSize": 23,
        "letterSpacing": -0.08,
      }
    }
  >
    Name
  </Text>
</ForwardRef>
`;
